src/accelerate/utils/modeling.py

```diff
@@ -929,6 +929,17 @@ def load_offloaded_weights(model, index, offload_folder):
         set_module_tensor_to_device(model, param_name, "cpu", value=weight, fp16_statistics=fp16_statistics)
 
 
+def get_module_leaves(module_sizes):
+    module_children = {}
+    for module in module_sizes:
+        if module == "":
+            continue
+        parent = module.rsplit(".", 1)[0]
+        module_children[parent] = module_children.get(parent, 0) + 1
+    leaves = [module for module in module_sizes if module_children.get(module, 0) == 0]
```

Comment:
Small nit even though it won't impact the logic in `get_balanced_memory`. 
```suggestion
    leaves = [module for module in module_sizes if module_children.get(module, 0) == 0 and module != ""]
```

Reply:
updated